x-pack/plugins/cases/server/common/models/case_with_comments.ts

```diff
@@ -245,8 +256,64 @@ export class CaseCommentModel {
     }
   }
 
+  private async filterDuplicatedAlerts(
+    attachments: CommentRequestWithId
+  ): Promise<CommentRequestWithId> {
+    const filterByIndices = (items: string[], indices: number[]): string[] =>
+      items.filter((_, itemIndex) => !indices.some((index) => index === itemIndex));
+
+    const filteredAlertAttachments: CommentRequestWithId = [];
+    const idsAlreadySeen = new Set();
+    const alertsAttachedToCase = await this.params.services.attachmentService.getter.getAllAlertIds(
+      {
+        caseId: this.caseInfo.id,
+      }
+    );
+
+    attachments.forEach((attachment) => {
+      if (!isCommentRequestTypeAlert(attachment)) {
+        filteredAlertAttachments.push(attachment);
+        return;
+      }
+
+      const { ids, indices } = getIDsAndIndicesAsArrays(attachment);
+      const indicesOfIdsAttachedToCase: number[] = [];
+
+      ids.forEach((id, index) => {
+        if (alertsAttachedToCase.has(id) || idsAlreadySeen.has(id)) {
+          indicesOfIdsAttachedToCase.push(index);
```

Comment:
Instead of recording the index in the array when an item is a duplicate, do you think it'd be more straightforward to make a copy of the ids and indices arrays and remove duplicate items as we iterate through?

It may be less performant but it might be more clear what is happening

Reply:
What about putting the id and the index that is not attached to the case to a structure like `{ id, index }` and then constructing the arrays from this structure? Will it make it clearer?

Reply:
@jonathan-buttner I went with renaming. Thanks for the help.